src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -5220,4 +5302,68 @@ namespace winrt::TerminalApp::implementation
 
         return profileMenuItemFlyout;
     }
+
+    void TerminalPage::_loadQueryExtension()
+    {
+        if (_extensionPalette)
+        {
+            return;
+        }
+
+        if (auto app{ winrt::Windows::UI::Xaml::Application::Current().try_as<winrt::TerminalApp::App>() })
+        {
+            if (auto appPrivate{ winrt::get_self<implementation::App>(app) })
+            {
+                // Lazily load the query palette components so that we don't do it on startup.
+                appPrivate->PrepareForAIChat();
+            }
+        }
+        _extensionPalette = winrt::Microsoft::Terminal::Query::Extension::ExtensionPalette();
+        _extensionPalette.RegisterPropertyChangedCallback(UIElement::VisibilityProperty(), [this](auto&&, auto&&) {
+            if (_extensionPalette.Visibility() == Visibility::Collapsed)
+            {
+                ExtensionPresenter().Visibility(Visibility::Collapsed);
+                _FocusActiveControl(nullptr, nullptr);
+            }
+        });
+        _extensionPalette.InputSuggestionRequested({ this, &TerminalPage::_OnInputSuggestionRequested });
+        _extensionPalette.ActiveControlInfoRequested([&](IInspectable const&, IInspectable const&) {
+            if (const auto activeControl = _GetActiveControl())
+            {
+                const auto profileName = activeControl.Settings().ProfileName();
+                const std::wstring fullCommandline = activeControl.Settings().Commandline().c_str();
+                const auto lastSlashPos = fullCommandline.find_last_of(L"\\");
+                if (lastSlashPos != std::wstring::npos)
+                {
+                    const auto end = fullCommandline.find_last_of(L"\"");
+                    const auto s = fullCommandline.substr(lastSlashPos + 1, end - lastSlashPos - 1);
+                    _extensionPalette.ActiveCommandline(fullCommandline.substr(lastSlashPos + 1, end - lastSlashPos - 1));
+                }
+                else
+                {
+                    _extensionPalette.ActiveCommandline(fullCommandline);
```

Comment:
For example: `cmd.exe /k #work 5` - there's no `\`, but the full commandline should probably just be `cmd.exe`, yea?